src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -159,26 +157,47 @@ - (void) setLearningRate:(float)lr {
   [_adamBeta setLearningRate:lr];
 }
 
-- (void) updateCurrentStyle:(NSUInteger)style {
+- (void) updateIndex:(NSUInteger)style {
   _currentStyle = style;
 }
 
 - (void) loadBeta:(float *)beta {
-  memcpy(_beta_weights.mutableBytes, beta, _numberOfFeatureChannels * _styles * sizeof(float));
+  float* betaWeights = (float *) [[[[style_props objectAtIndex: _currentStyle] betaVector] data] contents];
+  memcpy(betaWeights, beta, _numberOfFeatureChannels * _styles * sizeof(float));
 }
 
 - (float *) beta {
-  [self checkpointWithCommandQueue:_cq];
-  return (float *) [[[[style_props objectAtIndex: _currentStyle] betaVector] data] contents];
+  NSUInteger previousStyle = _currentStyle;
+  NSMutableData * betaPlaceHolder = [NSMutableData data];
+  for (NSUInteger index = 0; index < _styles; index++) {
+    _currentStyle = index;
+    [self checkpointWithCommandQueue:_cq];
+    float* betaWeights = (float *) [[[[style_props objectAtIndex: _currentStyle] betaVector] data] contents];
+    [betaPlaceHolder appendBytes:betaWeights length:sizeof(float)*_numberOfFeatureChannels];
+  }
+  _currentStyle = previousStyle;
+
+  return (float *) (betaPlaceHolder.bytes);
```

Comment:
This can return an invalid pointer, since `betaPlaceHolder` will go out of scope when this function returns. It may or may not work currently, depending on the behavior of the autorelease pool, but we should not write code that relies on this.

Reply:
How would you suggest we go about returning a float *. The MPS api requires this method to be overridden with a float * as the return type. Do we bypass their interface and create another method that returns with a type NSData?